redis/cache/redisx/redisxtest/redisxtest.go

```diff
@@ -0,0 +1,39 @@
+package redisxtest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/joomcode/redispipe/redis"
+	"github.com/joomcode/redispipe/redisconn"
+
+	"github.com/reddit/baseplate.go/redis/cache/redisx"
+)
+
+// NewMockRedisClient sets up a redis client for use in testing
+func NewMockRedisClient(
+	ctx context.Context,
+	tb testing.TB,
+	address string,
+	opts redisconn.Opts,
+) (*redisx.BaseSync, error) {
+	tb.Helper()
+
+	// Create connection to redis
+	conn, err := redisconn.Connect(ctx, address, opts)
+	if err != nil {
+		return nil, err
```

Comment:
If this is a testing helper, in my opinion it's better to call t.Fatalf and document that it must be called directly in a test / subtest, rather than do what is inevitable: have every user of this make their own _local_ helper that wraps _this_ helper that does the transition.  I prefix my t.Fatalf errors with "SETUP:" to make it clear that it's not a failure of the code under test, and I would name the function "Setup" not "New" to encourage that interpretation as well.